components/Cast/SpotifyEmbed.tsx

```diff
@@ -1,94 +1,79 @@
+  const [fullLoaded, setFullLoaded] = useState(false);
+
+  const togglePlay = () => {
+    if (!iframeRef?.current) return;
+    const spotifyEmbedWindow = iframeRef.current.contentWindow as any;
+    spotifyEmbedWindow.postMessage({ command: 'toggle' }, '*');
+  };
 
   useEffect(() => {
     const init = async () => {
-      if (!trackId) return;
-      const track = await getSpotifyTrack(trackId);
-      if (!('error' in track)) {
-        setTrack(track);
-        setDuration(track.duration_ms);
-      }
+      try {
+        const oEmbedUrl = `https://open.spotify.com/oembed?url=${encodeURIComponent(trackUrl)}`;
+        const response = await fetch(oEmbedUrl);
+        const data = await response.json();
+        const srcRegex = /src="([^"]+)"/;
+        const match = data.html.match(srcRegex);
+        const src = match ? match[1] : null;
+        setIframeSrc(src);
+        setEmbedData(data);
+        // eslint-disable-next-line no-empty
+      } catch (error) {}
     };
-
     init();
-  }, [trackId]);
+  }, [trackUrl]);
 
   useEffect(() => {
-    if (!(track?.uri && iframeApi)) return;
-
-    const options = {
-      height: '10',
-      width: '10',
-      uri: track.uri,
-    };
-    iframeApi.createController(elementRef.current, options, (embedController: any) => {
-      embedController.addListener('ready', () => {
-        setEmbedController(embedController);
-      });
-      embedController.addListener('playback_update', (e: SpotifyPlaybackUpdateEvent) => {
-        const data = e.data;
-        setPosition(data.position);
-        setDuration(data.duration);
-      });
+    if (!iframeRef?.current || !iframeSrc) return;
+    iframeRef.current.addEventListener('load', () => {
+      setFullLoaded(true);
     });
-  }, [track?.uri, iframeApi]);
+  }, [iframeRef, iframeSrc]);
 
-  if (track?.error) {
-    return <></>;
-  }
+  if (!embedData) return <></>;
 
   return (
     <div className="relative z-0 w-full">
       <MediaPlayer
         metadata={
-          track && {
-            id: track.uri,
+          embedData && {
+            id: trackUrl,
             type: 'spotify',
-            artistName: track.artists.map((artist: any) => artist.name).join(', '),
-            trackName: track.name,
-            artworkUrl: track.album.images[0].url,
-            duration,
+            artistName: '',
+            trackName: embedData.title,
+            artworkUrl: embedData.thumbnail_url,
+            duration: 0,
           }
         }
         controls={
-          fullLoadedEmbed
+          fullLoaded
             ? {
-                play: () => {
-                  embedController.togglePlay();
-                },
-                pause: () => {
-                  embedController.pause();
-                },
-                seek: (time) => {
-                  embedController.seek(time);
-                },
+                play: togglePlay,
+                pause: togglePlay,
```

Comment:
what happens if song has finished playing and I play a different song?
right now, seems like `play` will get triggered in the a`spotify` song as well as the other song